crates/containerd-shim-wasmtime/src/instance.rs

```diff
@@ -240,16 +221,23 @@ impl Wasi {
         let stdout = maybe_open_stdio(stdout).context("could not open stdout")?;
         let stderr = maybe_open_stdio(stderr).context("could not open stderr")?;
 
+        let wasmtime_executor = Box::new(WasmtimeExecutor {
+            engine,
+            stdin,
+            stdout,
+            stderr,
+        });
+        let default_executor = Box::new(LinuxContainerExecutor {
+            stdin,
+            stdout,
+            stderr,
+        });
         let container = ContainerBuilder::new(self.id.clone(), syscall.as_ref())
-            .with_executor(vec![Box::new(WasmtimeExecutor {
-                stdin,
-                stdout,
-                stderr,
-                engine,
-            })])?
+            .with_executor(vec![default_executor, wasmtime_executor])?
```

Comment:
hmm If we end up making the binary check optional on the builder (https://github.com/containers/youki/issues/2220) then it would mean the check wouldn't happen for linux containers as well.

Reply:
Wondering if the `can_handle` function on the executor is the correct place for that check? It looks like we are mostly checking the spec, to determine which executor to choose.  Another option might be to check and build the container with the right options?  that would solve "check for binary" problem.

Reply:
> Wondering if the can_handle function on the executor is the correct place for that check? 

I am wondering that too. At least my implementation of `can_handle` in the default executor is very similar to youki's verify_binary except that it also verifies if the executable is a Linux ELF file or a script file.

Reply:
If we make binary check optional on the builder, then the `can_handle` function will be used for checking the binary for linux containers. I think that's the right place for this check. 

It looks like the `Executor` trait and its `can_handle` function were removed from the main branch of `youki`.

Reply:
> It looks like the Executor trait and its can_handle function were removed from the main branch of youki

😮 interesting, does this change anything?

Reply:
I assume there needs some work to refactor to the latest change of youki. #209

Reply:
Sorry for confusing you. Don't worry.
https://github.com/containers/youki/issues/2220#issuecomment-1676356315